tests/torch/modules/test_rnn.py

```diff
@@ -623,7 +623,7 @@ def hook(model, input_, counter):
         new_seq_len = int(sequence_size / 2)
         dummy_forward_fn(model, new_seq_len)
         # NB: may always fail in debug due to superfluous 'cat' nodes
-        assert model.nncf.get_graph().get_nodes_count() == 373
+        assert model.nncf.get_graph().get_nodes_count() == 370
```

Comment:
![image](https://github.com/openvinotoolkit/nncf/assets/4014476/271ff8bb-b0d6-4654-90eb-e8ae22e00879)
Fewer nodes are considered within iteration with more strict condition. 
E.g. `GNMT/ResidualRecurrentEncoder[encoder]/Dropout[dropout]/dropout_2` doesn't correspond to any of the registered iteration modules: 
![image](https://github.com/openvinotoolkit/nncf/assets/4014476/d24ec235-4c88-4b7e-8d97-7586c570e621)